### PR TITLE
FIX: agdaPackages.cubical

### PR DESCRIPTION
**Pull request by guilhermehas**
###### Motivation for this change

Cubical Agda library was not working in Agda version 2.6.2. So I fixed it changing it to a newer version.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
you need to change the `version = "..."` to unstable and the date of the commit 
i made this issue so maybe we will get a release https://github.com/agda/cubical/issues/583

File: pkgs/development/libraries/agda/cubical/default.nix [hunk 1]
```diff
@@ -9,12 +9,14 @@ mkDerivation rec {
...
     owner = "agda";
-    rev = "v${version}";
-    sha256 = "07qlp2f189jvzbn3aqvpqk2zxpkmkxhhkjsn62iq436kxqj3z6c2";
+    rev = "d5030a9c89070255fc575add4e9f37b97e6a0c0c";
+    sha256 = "8WIrgAyA0bwFgtEjmwlvKQ7rZB91/Ppi8zOSq/qCTKE=";
   };
 
   LC_ALL = "en_US.UTF-8";
 
+  patches = [ ./no_werror_patch.txt ];
```
Comment (Mic92) [inline pkgs/development/libraries/agda/cubical/default.nix:18]: This should have the same effect:

```
preConfigure = ''makeFlagsArray=("AGDA_EXEC=agda -W error")'';
```

**Comment by alexarice** [comment]
We usually wait for a release before updating agda libraries but Its probably preferable to at least have something that works

File: pkgs/development/libraries/agda/cubical/default.nix (hunk 1 above)
Comment (alexarice) [reply pkgs/development/libraries/agda/cubical/default.nix:18]: This would need to be the other way round I think

Comment (guilhermehas) [reply pkgs/development/libraries/agda/cubical/default.nix]: I have already fixed it.

**Comment by guilhermehas** [comment]
I changed the version to unstable.

**Comment by Artturin** [comment]
Use git rebase to fixup the commits into the first commit and reword the first commit 
https://git-rebase.io/#fixup
Discard the `deleted tested package` And `returned all-packages.nix`

**Comment by guilhermehas** [comment]
> Discard the `deleted tested package` And `returned all-packages.nix`
> 
> Use git rebase to fixup the commits into the first commit and reword the first commit
> https://git-rebase.io/#fixup

I have already done the fixup. Look if everything is right.

**Comment by Artturin** [comment]
```
reword 30a0c213eb4 added newer version of cubical Agda
fixup 6bc702b4fa9 added patch to cubical agda
fixup a2f61df7776 agdaPackages.cubical: 0.2 -> unstable
fixup 82034c77c7f agdaPackages.cubical: 0.2 -> unstable
drop cb1e38b389e deleted tested package
drop 548aedd729f returned all-packages.nix
```
in the next dialog
```
agdaPackages.cubical: 0.2 -> unstable
```

it would have been like this but this pr seems to be not needed now https://github.com/NixOS/nixpkgs/pull/131226

Thanks you for the pr though!

**Comment by guilhermehas** [comment]
new pull request:
https://github.com/NixOS/nixpkgs/pull/131280

**Pull request closed without merging**